include/swift/AST/DiagnosticsParse.def

```diff
@@ -1356,6 +1358,13 @@ ERROR(expected_type_after_as,none,
 ERROR(string_interpolation_extra,none,
       "extra tokens after interpolated string expression", ())
 
+// String interpolation isnt closed by a )
+// ie: `let west = "ye \("`
+ERROR(string_interpolation_unclosed, none,
+      "expected ')' at end of string interpolation", ())
+NOTE(string_interpolation_unclosed_fix_it, none,
+      "add ')' to end of interpolation statement", ())
+
```

Comment:
Since the fix-it is trivial and alluded to in the error message, should we attach it directly to the error rather than a separate note? Xcode's inline diagnostics don't really care, but it would make things more straightforward and less repetitive when displayed in the terminal.

Reply:
Are you sure? The note looks fine to me and makes sense.. @jckarter @hamishknight what do you guys think?

Reply:
Okay I think I misread what you said sorry lol, but if we were to merge both into one, what would you word it as? And I thought swift would always seperate fix-its into seperate notes

Reply:
> I thought swift would always separate fix-its into separate notes

We do use separate notes for multiple fix-its, or a single fix-it that warrants a custom message describing the implications, i.e. force-unwrapping. "did you mean/consider" diagnostics, on the other hand, usually have a fix-it attached directly.

> what would you word it as? 

Xcode generates default descriptions for range replacements, `Insert ')'` or similar in this case.

Reply:
Good idea, I think I’ll word it as “Insert ‘)’ to end interpolation statement expression”

Reply:
To be clear, what I meant is that you don't need a wording (you can't have one separately as with a note anyway) *if* you choose to attach the fix-it to the error instead, because Xcode provides a default description for you.

Reply:
Oh I see, what about in the terminal then? Will the fix-it be shown on the same line as the error text?

Reply:
The only difference will be the absence of a note:

```
test.swift:1:13: error: expected ')' at end of string interpolation
let a = "\("
           ^
           )
```

Reply:
I see, I’ll change it to just the error then (once I get it actually working properly..)

Reply:
About the note to match the beginning paren: that’s a great idea, I guess that would be easy to do with the current code structure